apps/roam/src/components/settings/DiscourseNodeConfigPanel.tsx

```diff
@@ -40,6 +49,60 @@ const DiscourseNodeConfigPanel: React.FC<DiscourseNodeConfigPanelProps> = ({
     }
   };
 
+  const handleDeleteNodeTypeWithConfirmation = (
+    nodeTypeIdToDelete: string,
+    nodeLabel: string,
+  ) => {
+    const affectedRelations = getDiscourseRelations().filter(
+      (r) =>
+        r.source === nodeTypeIdToDelete || r.destination === nodeTypeIdToDelete,
+    );
+
+    let dialogMessage = `Are you sure you want to delete the Node Type "${nodeLabel}"?`;
+
+    if (affectedRelations.length > 0) {
+      dialogMessage = `The Node Type "${nodeLabel}" is used by the following relations, which will also be deleted:\n\n${affectedRelations
+        .map((r) => {
+          const sourceNodeDetails = nodes.find((s) => s.type === r.source);
+          const destinationNodeDetails = nodes.find(
+            (d) => d.type === r.destination,
+          );
+          return `- ${sourceNodeDetails?.text || r.source} ${r.label} ${destinationNodeDetails?.text || r.destination}`;
+        })
+        .join("\n")}\n\nProceed with deletion?`;
+    }
+
+    setAlertMessage(dialogMessage);
+    setAlertConfirmAction(() => async () => {
+      try {
+        for (const rel of affectedRelations) {
+          await deleteBlock(rel.id).catch((error) => {
+            console.error(
+              `Failed to delete relation: ${rel.id}, ${error.message}`,
+            );
+            throw error;
+          });
+        }
+        await window.roamAlphaAPI.deletePage({
+          page: { uid: nodeTypeIdToDelete },
+        });
+
+        setNodes((prevNodes) =>
+          prevNodes.filter((nn) => nn.type !== nodeTypeIdToDelete),
+        );
+        refreshConfigTree();
+        setDeleteConfirmation(null);
+      } catch (error) {
+        console.error(
+          `Failed to complete deletion for Node Type ${nodeLabel} (UID: ${nodeTypeIdToDelete}): ${error instanceof Error ? error.message : String(error)}`,
+        );
+      } finally {
+        setIsAlertOpen(false);
+      }
+    });
+    setIsAlertOpen(true);
```

Comment:
Currently this is adding an additional confirmation regardless if we need to delete relations or not.

We should only require the additional confirmation IF there are relations to delete.